systemd/cloud-final.service.tmpl

```diff
@@ -15,7 +19,16 @@ ExecStart=/usr/bin/cloud-init modules --mode=final
 RemainAfterExit=yes
 TimeoutSec=0
 KillMode=process
+{% if variant == "rhel" %}
+# Restart NetworkManager if it is present and running.
+ExecStartPost=/bin/sh -c 'u=NetworkManager.service; \
+ out=$(systemctl show --property=SubState $u) || exit; \
+ [ "$out" = "SubState=running" ] || exit 0; \
+ systemctl reload-or-try-restart $u'
```

Comment:
This is an interesting behavior in cloud-final as we should have earlier either emitted the proper network prior to NetworkManager.service first bringup for datasources which support discovery in init-local stage. Or cloud-init should have restarted network in 'init(-network)' stage for datasources detected in the init-network stage just after we write network config. I'm wondering what failure mode we are trying to work around here.

Reply:
The original bug is this one: https://bugzilla.redhat.com/show_bug.cgi?id=1897616
Apparently we are covering the case where NM is not installed

Reply:
Ok this works for me. Thanks for the additional context and looks like it's been in streams for a while too! +1